### ISSUE TITLE
[BUG] Project monitor status got stuck at "WaitingForDashboardValues".

### ISSUE THREAD

**Issue by pankajsqe** [rancher/rancher#47155]
**Rancher Server Setup**
- Rancher version: v2.10-head
- Installation option (Docker install/Helm Chart): Docker
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): K3S

**Information about the Cluster**
- Cluster Type (Local/Downstream): Local

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom) : Admin

**Describe the bug**
While the installing Prometheus Federator 105.0.0-rc.1+up0.4.2 on Rancher 2.10-head via Docker I encountered a problem when setting up the project monitor. The setup process is stuck with the status "WaitingForDashboardValues"

Additionally, the corresponding pod, helm-install-cattle-project-p-7mj9v-monitoring-jwr97, is throwing the following error:

`Error: INSTALLATION FAILED: rendered manifests contain a resource that already exists. Unable to continue with install: Namespace "cattle-dashboards" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: key "meta.helm.sh/release-name" must equal "cattle-project-p-7mj9v-monitoring": current value is "rancher-monitoring"; annotation validation error: key "meta.helm.sh/release-namespace" must equal "cattle-project-p-7mj9v-monitoring": current value is "cattle-monitoring-system"`

**Details**

- rancher v2.10-head
- cluster k3s
- prom-fed 105.0.0-rc.1+up0.4.2
- charts repo :
```
type: git
url: https://github.com/mallardduck/rancher-charts
branch: prom-reorg-test
```

**To Reproduce**
Steps:

- Install Rancher using the Docker image rancher/rancher:head (2.10-head).
- Deploy the latest Rancher Monitoring chart under the system project.
- Update the charts repository with the following details and then deploy the Prometheus Federator chart version 105.0.0-rc.1+up0.4.2:
   Type: git
   URL: https://github.com/mallardduck/rancher-charts
   Branch: prom-reorg-test
- Create a new project and corresponding namespace (e.g., test-prom-fed and test-prom-fed-ns).
- Navigate to Monitoring -> Project Monitor and create a new project monitor.
- From the dropdown, select the project you created (e.g., test-prom-fed).
- Observe that the status is displayed as “WaitingForDashboardValues.”
- Check the workload for the failing pod helm-install-cattle-project.

**Result**
Project monitor setup process got stuck with the status "WaitingForDashboardValues"

**Expected Result**
project monitor should be created successfully.

**Screenshots**
<img width="1728" alt="Screenshot 2024-09-16 at 7 43 48 PM" src="https://github.com/user-attachments/assets/d364ec32-db71-45b4-8520-4c5a9674a492">
<img width="1728" alt="Screenshot 2024-09-16 at 7 43 43 PM" src="https://github.com/user-attachments/assets/943ece51-85ec-4c22-872c-89594f70010c">
<img width="1728" alt="Screenshot 2024-09-16 at 7 42 52 PM" src="https://github.com/user-attachments/assets/93e64890-c7f5-42d3-b7d5-6c482dc84754">

**Comment by mallardduck** [comment]
So this issue goes back further than my changes do. I've been able to replicate the issue on the following setup:

Rancher: `v2.10-a1a0d2edf04548b0a9099f86bcf8c194771db8f8-head`
k8s: k3s - `v1.27.7+k3s2`
Storage: 	`longhorn:104.1.0+up1.6.2`1
Monitoring: `rancher-monitoring:105.1.0-rc.4+up61.3.2`
Federator: `prometheus-federator:104.0.1+up0.4.2`
Workload: a random pihole chart (doesn't matter but used a real one for extra realism)

And I see:
<img width="1461" alt="Screenshot 2024-09-20 at 10 15 15 AM" src="https://github.com/user-attachments/assets/d93c2357-6d31-424c-afa7-2e73c1b4b3ff">

**Comment by mallardduck** [comment]
This seems to be a potential incompatibility between `rancher-monitoring` and the `rancher-project-monitoring` chart that gets deployed. In my fork/branch it is deploying from `0.4.2` chart version and the last working version of the chart (in my testing) is the `0.3.x` versions of the chart.

 I’ve found that starting in 0.4.0 the chart started to include the namespace as part of the charts templates file. And by adding `.grafana.defaultDashboards.useExistingNamespace` and setting it to `true` this resolves the currently reported bug. However there are additional errors I'm seeing still that I'm suspecting may be related to `0.4.x` versus `0.3.x` chart differences in general.

**Comment by mallardduck** [comment]
To enable easier debugging I've produced images with the following tags:

- `mallardduck/prometheus-federator:reorg-v0.3.3`
- `mallardduck/prometheus-federator:reorg-v0.3.4`
- `mallardduck/prometheus-federator:reorg-v0.4.0`
- `mallardduck/prometheus-federator:reorg-v0.4.1`
- `mallardduck/prometheus-federator:reorg-v0.4.2` - a rebuild/retag of the `
reorg-demo` image used by `prom-reorg-test` branch chart by default.

Each version suffix correlates to the version of the `rancher-project-monitoring` chart that is embedded into the `prometheus-federator` binary/image.

In order to test images individually without resetting everything, it's possible to edit the `prometheus-federator` App.

- Under edit/upgrade, click "Edit YAML", then
- adjust the `.helmProjectOperator.image.tag` to match desired target.
- Verify the deployment updates to the new image first.
- Then proceed to delete the existing project monitor, and 
- then create a new one for it with the new image.

**Comment by mallardduck** [comment]
TL;DR: We were not embedding newer versions of `rancher-project-monitoring` into the `prometheus-federator` images and as such all prior versions are using [0.3.4](https://github.com/rancher/prometheus-federator/tree/main/charts/rancher-project-monitoring/0.3.4).

Summary:
If you install `promeheus-federator` then create a Project monitor it will create a `HelmChart`, you can then extract the original chart from this manifest. After extracting the chart contents and then decompressing them you can review the chart version in use. In my testing on friday, while easily confused about many components with similar versions, I did ultimately discover that the version of rancher-project-monitoring was not connected 1:1 with the version of prometheus-federator. In turn, the `0.4.x` versions of Prometheus Federator embed the `0.3.x` chart for `rancher-project-monitoring`.

**Comment by mallardduck** [comment]
@pankajsqe - The branch has been updated and is ready for re-testing to validate the PR. Either refresh your App repo, if you have the existing cluster, or re-test on a new cluster and it should grab the fresh files from my fork of `rancher/charts`.

**Comment by pankajsqe** [comment]
Retest-Passed : Project monitor status got updated from "WaitForDashboardValues" to "Deployed".

Details

rancher v2.10-head
cluster k3s
prom-fed 105.0.0-rc.1+up0.4.2
charts repo :
```
type: git
url: https://github.com/mallardduck/rancher-charts
branch: prom-reorg-test
```

![Image](https://github.com/user-attachments/assets/65367564-0d72-4bae-9d8a-8d9ff5bb6476)
![Image](https://github.com/user-attachments/assets/4c26a0fd-deae-4886-a958-1202f0ad59a1)
